packages/terra-search-field/src/SearchField.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconSearch from 'terra-icon/lib/icon/IconSearch';
+import classNames from 'classnames';
+import './SearchField.scss';
+
+const propTypes = {
+  /**
+   * Placeholder text to show while the search field is empty.
+   */
+  placeholder: PropTypes.string,
+
+  /**
+   * The minimum number of characters to perform a search.
+   */
+  minimumSearchTextLength: PropTypes.number,
+
+  /**
+   * How long the component should wait after input before performing an automatic search.
+   */
+  searchDelay: PropTypes.number,
+
+  /**
+   * A callback to perform search. Sends parameter {String} searchText.
+   */
+  onSearch: PropTypes.func,
+};
+
+const defaultProps = {
+  placeholder: '',
+  minimumSearchTextLength: 2,
+  searchDelay: 250,
+};
+
+class SearchField extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.handleTextChange = this.handleTextChange.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+
+    this.searchTimeout = null;
+    this.state = {
+      searchText: '',
+    };
+  }
+
+  handleTextChange(event) {
+    this.setState({ searchText: event.target.value });
+
+    if (!this.searchTimeout) {
+      this.searchTimeout = setTimeout(this.handleSearch, this.props.searchDelay);
+    }
+  }
+
+  handleSearch() {
+    this.clearSearchTimeout();
+
+    if (this.state.searchText.length >= this.props.minimumSearchTextLength && this.props.onSearch) {
+      this.props.onSearch(this.state.searchText);
+    }
+  }
+
+  clearSearchTimeout() {
+    if (this.searchTimeout) {
+      clearTimeout(this.searchTimeout);
+      this.searchTimeout = null;
+    }
+  }
+
+  render() {
+    const { placeholder, ...customProps } = this.props;
+    const searchFieldClassNames = classNames(['terra-SearchField',
+      customProps.className,
+    ]);
+
+    return (
+      <div className={searchFieldClassNames}>
```

Comment:
Custom props should be spread onto the container here, before className. This allows consumers to pass through any extra html properties that aren't supplied first class through the prop API. ( ID, className, data attributes, etc.. )

Examples:
https://github.com/cerner/terra-core/blob/master/packages/terra-progress-bar/src/ProgressBar.jsx#L49
https://github.com/cerner/terra-core/blob/master/packages/terra-status/src/Status.jsx#L24